streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -86,6 +87,10 @@ public boolean onlyStandbyTasksLeft() {
             return !updatingTasks.isEmpty() && updatingTasks.values().stream().noneMatch(Task::isActive);
         }
 
+        public Collection<Task> getPausedTasks() {
```

Comment:
Just to confirm this will not change the pause resume API?

Reply:
No, I just moved the map of paused tasks to the state updater thread. The state updater does not have permissions to access private fields of the state udater thread due to encapsulation, so I needed to add a getter for the tasks. The behavior stays the same. You can see below in `getPausedTasks()` that the call to `pausedTasks.values()` was replaced with a call to this method which does exactly the same.